tests/unit/task/task_builder_test.py

```diff
@@ -182,7 +183,7 @@ async def test_failing_decorator_continues(
         task_config.exception_handler.assert_not_called()
 
     @pytest.mark.asyncio
-    async def test_decorator_variables_are_added(
```

Comment:
I think we should split this test into `test_decorator_variables_are_added` and `test_job_status_is_updated`.

This way we can know exactly why the tests have failed.

Reply:
Thanks for the comment @JonatanMartens  :) I have actioned and fixed up my commits. Please re-review